### PR TITLE
Feat: mark ranges without graph; Closes #1508

### LINKED ISSUE #1508: Add current XP value for each Mark Range to the Mark Calculations Page
Currently, Mark Ranges are only shown on the graph, but it is too hard to figure out what the current XP value needed to achieve the % needed to hit a Mark Range.  You can hover over the chart but it is awkward:
image

There should be a section added to the Mark Calculations Page that shows the current minimum XP value (rounded to integer) required to hit each of the mark range.

This could be added below the chart on the XP Progress tab.

### PR DESCRIPTION
**Pull request by 9921622**
Please ensure you are familiar with our Pull Request Description expectations described here: https://www.pullrequest.com/blog/writing-a-great-pull-request-description/
### What?
Added a new section to ``` /courses/marks ``` called "Current Mark Ranges by XP"

### Why?
See #1508

### How?

inside view added xp_needed variable to the markranges. Used the xp formula inside ```xp_chart_progress.html``` as reference
image
xp formula 
image

inside html I used the xp_needed to display the format specified in #1508.
image

### Testing?
Added a new test class for mark ranges.
I did originally add this to the courseview test cases. But had a lot of issues with mock patch not working inside that test case
image
image
image

### Screenshots (if front end is affected)
image

### Review request
@tylerecouture

## Summary by CodeRabbit

- **New Features**
  - Added a section to display current mark ranges by XP in the mark calculations view.
  - Introduced new tests to ensure correct display of mark ranges by XP for different completion levels of the semester.

- **Bug Fixes**
  - Fixed the calculation and display of XP needed for passing mark ranges based on minimum marks and completion percentage.

### CODE REVIEW

**Comment by 9921622** [comment]
@tylerecouture review

**Comment by 9921622** [comment]
It might actually be best to put the test case in CourseView test if i somehow fix it. Submitting this pr just for a second opinion

File: src/courses/templates/courses/mark_calculations.html [hunk 1]
```diff
@@ -118,6 +118,15 @@ <h3>Explanation</h3>
         </li>
       </ul>
 
+      <h3>Current Mark Ranges by XP</h3>
```
Comment (tylerecouture) [inline src/courses/templates/courses/mark_calculations.html:121]: This should be below the charts:
image

File: src/courses/views.py [hunk 2]
```diff
@@ -65,6 +65,14 @@ def mark_calculations(request, user_id=None):
     # combine assigned_ranges and all_ranges, then order by min mark and only include markranges with unique minimum marks in context queryset
     markranges = (assigned_ranges | all_ranges).order_by('minimum_mark').distinct('minimum_mark')
 
+    # inject the xp needed for passing the mark range
+    # cant do it with template tags as you can only multiply/divide once
+    days_percentage = course_student.semester.fraction_complete()
+    total_xp = courses.first().course.xp_for_100_percent
+    for markrange in markranges:
+        mark_percentage = markrange.minimum_mark / 100
+        markrange.xp_needed = round(total_xp * mark_percentage * days_percentage)
```
Comment (tylerecouture) [inline src/courses/views.py:74]: `Round` is not the same as `Floor` (thank you for providing the reference in your PR notes!).  These will give different results for 34.6  one will be 34 and the other will be 35.  Please use Floor or round down in your python too.

**Review by tylerecouture — CHANGES_REQUESTED** [review]

File: src/courses/tests/test_views.py [hunk 3]
```diff
@@ -1048,3 +1048,67 @@ def test_no_test_users_in_histogram_values(self):
...
+        self.assertContains(response, '641')  # 750 * 0.855 = 641.25
+
+        # Check if markranges show as 100% of 1000 xp
+        mock_sem_fraction_complete.return_value = 1
+        response = self.client.get(reverse('courses:my_marks'))
+
+        self.assertEqual(1000 * mock_sem_fraction_complete.return_value, 1000)
+        self.assertContains(response, '495')  # 1000 * 0.495 = 495
+        self.assertContains(response, '725')  # 1000 * 0.725 = 725
+        self.assertContains(response, '855')  # 1000 * 0.855 = 855
```
[the remark this answers is not in the document]
Comment (9921622) [reply src/courses/tests/test_views.py:1114]: ok

File: src/courses/templates/courses/mark_calculations.html (hunk 1 above)
Comment (9921622) [reply src/courses/templates/courses/mark_calculations.html:121]: done

File: src/courses/views.py (hunk 2 above)
Comment (9921622) [reply src/courses/views.py:74]: done

**Comment by 9921622** [comment]
ok. added all changes, and fixed an error regarding the view code that applied to tests outside ```courses```. Specifically ```profile_manager```

**Review by tylerecouture — APPROVED** [review]

**Pull request merged**